tests/basic/telemetry_test.go

```diff
@@ -233,77 +233,93 @@ var _ = Describe("{ProfileOnlyDiags}", func() {
+			pxInstalled, err = Inst().V.IsPxInstalled(currNode)
+			if err != nil {
+				logrus.Debugf("Could not get PX status on %s", currNode.Name)
+			}
+			if pxInstalled {
+				// Get the most recent profile diags for comparison
+				Step(fmt.Sprintf("Check latest profile diags on node %v", currNode.Name), func() {
+					logrus.Infof(" Getting latest profile  diags on %v", currNode.Name)
+					existingDiags, err = telemetryRunCmd(fmt.Sprintf("ls -t /var/cores/%s-*.{stack,heap}.gz | head -n 2", currNode.Name),
 						currNode, nil)
 					if err == nil {
-						if existingDiags != "" && existingDiags == newDiags {
-							logrus.Infof("No new profile diags found...")
-							newDiags = ""
-						}
-						if len(newDiags) > 0 {
-							logrus.Infof("Found new profile diags:\n%s", newDiags)
-							// Needs to contain both stack/heap
-							if strings.Contains(newDiags, ".heap") && strings.Contains(newDiags, ".stack") {
-								diagsFiles = strings.Split(newDiags, "\n")
-								return true
-							}
-						}
+						logrus.Infof("Found latest profiles diags on node %s:\n%s ", currNode.Name, existingDiags)
+					} else {
+						existingDiags = ""
 					}
-					return false
-				}, 5*time.Minute, 15*time.Second).Should(BeTrue(), "failed to generate profile diags on node %s", currNode.Name)
-			})
-			Step(fmt.Sprintf("Validate diags uploaded on S3"), func() {
-				for _, file := range diagsFiles {
-					fileNameToCheck := path.Base(file)
-					logrus.Debugf("Validating file %s", fileNameToCheck)
-					if TelemetryEnabled(currNode) { // This is done in case the system is run without telemetry.
-						err := Inst().V.ValidateDiagsOnS3(currNode, fileNameToCheck)
+				})
+				// Issue a profile diags to generate new files.
+				Step(fmt.Sprintf("collect profile diags on node: %s | %s", currNode.Name, currNode.Type), func() {
+					config := &torpedovolume.DiagRequestConfig{
+						DockerHost: "unix:///var/run/docker.sock",
+						Profile:    true,
+					}
+					if TelemetryEnabled(currNode) {
+						err := Inst().V.CollectDiags(currNode, config, torpedovolume.DiagOps{Validate: true})
 						if err != nil {
-							logrus.Errorf("Failed to validate diags: %v", err)
+							logrus.Errorf("Failed to collect diags on Node: %s Err: %v", currNode.Name, err)
 						}
-						Expect(err).NotTo(HaveOccurred(), "Files validated on s3")
+						Expect(err).NotTo(HaveOccurred(), "Profile only diags collected successfully")
 					} else {
-						logrus.Debugf("Telemetry not enabled on %s, skipping test", currNode.Name)
+						logrus.Infof("Telemetry not enabled on the node %s", currNode.Name)
+						skipTest = true
+						return
+					}
+				})
+				// Get the latest files in the directory.  The newly generated files will not equal the most recent. So you know they are new.
+				Step(fmt.Sprintf("Get the new profile diags on node %v", currNode.Name), func() {
+					if skipTest {
```

Comment:
I don't think you need this check 'if skipTest {'? The reason being is the check is being done in JustBeforeEach(func() { above which should skip the whole test.

Reply:
So discussed with @aghodke1312 ... The intent is to have the test run all the time so we need to remove the `TelemetryEnabled` check in  `JustBeforeEach(func() {`.